### DISCUSSION TITLE
Storage layers + permissions

### DISCUSSION THREAD

**Discussion by nlathia** [operatorai/modelstore#152] [Ideas]
There's a use case for `modelstore` where the storage permissions vary by user. The full set of possibilities are:

The full set of actions that are currently supported are:

- Write: upload a model, create, set a model state
- Read: download / load a model, get a model state, get domain information
- Delete: delete a model, remove a model state

But, we can assume that different users of `modelstore` may have varying permissions, starting with read-only ones:

- https://github.com/operatorai/modelstore/pull/142

In the longer term, it would be great if:

1. We have feature parity across the different storage types - e.g., allowing anonymous access to public s3 buckets;
2. Can - where possible - automatically figure out which permissions are available on the selected storage type

**Comment by dbuades** [comment]
This would definitely be very useful!
